### PR TITLE
libfprint-2-tod1-goodix-550a: init at 0.0.9

### PR DESCRIPTION
**Pull request by utkarshgupta137**
###### Description of changes

Added the driver for Goodix 27c6::550a fingerprint reader, provided by Lenovo.
Even though the driver is provided "only" for Thinkpad laptops, it is working perfectly fine on my Lenovo Legion (16ARHA7).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  src = fetchurl {
+    url = "https://download.lenovo.com/pccbbs/mobiles/r1slg01w.zip";
+    sha256 = "sha256-hg4h7cV88Tmecucf1B4N73Y5z+hJgm6+TgSSESvJ2Jc=";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  unpackPhase = ''
+    unzip $src
+    unzip libfprint-tod-goodix-550a-0.0.9.zip
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix:16]: Can't we use fetchzip?

File: pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  passthru.driverPath = "/lib/libfprint-2/tod-1";
+
+  meta = with lib; {
+    description = "Goodix 550a driver module for libfprint-2-tod Touch OEM Driver (from Lenovo)";
+    homepage = "https://support.lenovo.com/us/en/downloads/ds560884-goodix-fingerprint-driver-for-linux-thinkpad-e14-gen-4-e15-gen-4";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    license = licenses.unfree;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ utkarshgupta137 ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix:44]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

File: pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix (hunk 1 above)
Comment (utkarshgupta137) [reply pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix]: Done

File: pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix (hunk 2 above)
Comment (utkarshgupta137) [reply pkgs/development/libraries/libfprint-2-tod1-goodix-550a/default.nix:43]: Done

**Pull request merged**